### PR TITLE
chore: move legacy api endpoints to async ingestion

### PR DESCRIPTION
**Pull request by Steffen911**
## What

Makes the 
- traces
- scores
- generations
- events
- span

endpoints use the new wrapper around our async ingestion pipeline instead of handling them synchronously.

----

> [!IMPORTANT]
> Refactor legacy API endpoints to use async ingestion pipeline, replacing synchronous processing with `processEventBatch`.
> 
>   - **Behavior**:
>     - Refactor `events.ts`, `generations.ts`, `spans.ts`, `scores/index.ts`, and `traces/index.ts` to use `processEventBatch` for async ingestion.
>     - Remove synchronous handling functions like `transformEventToIngestionBatch` and `handleBatch`.
>     - Add error handling for async processing failures, returning appropriate HTTP status codes.
>   - **Functions**:
>     - Add `processEventBatch` in `ingestion.ts` to handle async event processing.
>     - Remove `parseSingleTypedIngestionApiResponse` from `ingestion.ts`.
>   - **Misc**:
>     - Update logging and error messages to reflect async processing in `events.ts`, `generations.ts`, `spans.ts`, `scores/index.ts`, and `traces/index.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=langfuse%2Flangfuse&utm_source=github&utm_medium=referral)<sup> for b1c11968f78a485850d30eb35bae8bfb9267a3cc. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: web/src/pages/api/public/events.ts [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
+        res
+          .status(error.status)
+          .json({ message: error.error ?? error.message });
+        return { id: "" }; // dummy return
+      }
+      if (result.successes.length !== 1) {
+        logger.error("Failed to create score", { result });
+        throw new Error("Failed to create score");
+      }
+      return result.successes[0];
```
Comment (Steffen911) [inline web/src/pages/api/public/events.ts:41]: @maxdeichmann This part seems a bit repetitive across all the endpoint implementations. Given that we want to remove most of them eventually, I prefer the verbosity with explicit handling over a utility function that abstracts and hides what happens. Let me know if you have another preference.

File: web/src/pages/api/public/events.ts [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+      if (!event.body.id) {
+        event.body.id = v4();
+      }
+      const result = await processEventBatch([event], auth);
+      if (result.errors.length > 0) {
+        const error = result.errors[0];
+        res
+          .status(error.status)
+          .json({ message: error.error ?? error.message });
+        return { id: "" }; // dummy return
```
Comment (maxdeichmann) [inline web/src/pages/api/public/events.ts:35]: why is this needed? Cant you just return res
          .status(error.status)
          .json({ message: error.error ?? error.message });

File: web/src/pages/api/public/ingestion.ts [hunk 3]
```diff
@@ -549,47 +549,3 @@ export const aggregateBatchResult = (
...
-
-  if (results.length === 0) {
-    throw new InternalServerError("No results returned");
-  }
-
-  if (object === undefined) {
-    return results[0].result as T extends z.ZodTypeAny ? z.infer<T> : unknown;
-  }
-
-  const parsedObj = object.safeParse(results[0].result);
```
Comment (maxdeichmann) [inline web/src/pages/api/public/ingestion.ts:588]: i think all the errors here above return different http status codes. Anything against using parseSingleTypedIngestionApiResponse?

Comment (Steffen911) [reply web/src/pages/api/public/ingestion.ts:588]: @maxdeichmann We should already cover the status codes via https://github.com/langfuse/langfuse/blob/main/web/src/pages/api/public/ingestion.ts#L491 and the error.status property. IMO the parseSingleTypedIngestionApiResponse is fully covered by aggregateBatchResults.
